src/test/javascript/spec/component/shared/navbar.component.spec.ts

```diff
@@ -0,0 +1,494 @@
+            } as MockBreadcrumb;
+
+            expect(component.breadcrumbs.length).to.equal(5);
+
+            // Use matching here to ignore non-semantic differences between objects
+            sinon.assert.match(component.breadcrumbs[0], courseManagementCrumb);
+            sinon.assert.match(component.breadcrumbs[1], testCourseCrumb);
+            sinon.assert.match(component.breadcrumbs[2], programmingExercisesCrumb);
+            sinon.assert.match(component.breadcrumbs[3], { label: 'Test Exercise', translate: false, uri: '/course-management/1/programming-exercises/2/' } as MockBreadcrumb);
+            sinon.assert.match(component.breadcrumbs[4], assessmentCrumb);
+        });
+
+        it('programming exercise hints', () => {
+            const testUrl = '/course-management/1/exercises/2/hints/3';
+            router.setUrl(testUrl);
+
+            const hintService = fixture.debugElement.injector.get(ExerciseHintService);
+            const hintsStub = sinon.stub(hintService, 'find').returns(of({ body: { title: 'Exercise Hint' } as ExerciseHint } as HttpResponse<ExerciseHint>));
+
+            fixture.detectChanges();
+
+            expect(courseManagementStub).to.have.been.calledWith(1);
+            expect(exerciseStub).to.have.been.calledWith(2);
+            expect(hintsStub).to.have.been.calledWith(3);
+
+            const hintsCrumb = {
+                label: 'artemisApp.exerciseHint.home.title',
+                translate: true,
+                uri: '/course-management/1/exercises/2/hints/',
+            } as MockBreadcrumb;
+
+            const hintCrumb = {
+                label: 'Exercise Hint',
+                translate: false,
+                uri: '/course-management/1/exercises/2/hints/3/',
+            } as MockBreadcrumb;
+
+            expect(component.breadcrumbs.length).to.equal(6);
+
+            // Use matching here to ignore non-semantic differences between objects
+            sinon.assert.match(component.breadcrumbs[0], courseManagementCrumb);
+            sinon.assert.match(component.breadcrumbs[1], testCourseCrumb);
+            sinon.assert.match(component.breadcrumbs[2], { label: 'artemisApp.course.exercises', translate: true, uri: '/course-management/1/exercises/' } as MockBreadcrumb);
+            sinon.assert.match(component.breadcrumbs[3], { label: 'Test Exercise', translate: false, uri: '/course-management/1/exercises/2/' } as MockBreadcrumb);
+            sinon.assert.match(component.breadcrumbs[4], hintsCrumb);
+            sinon.assert.match(component.breadcrumbs[5], hintCrumb);
+        });
+
+        it('text exercise feedback conflict', () => {
+            const testUrl = '/course-management/1/text-exercises/2/submissions/3/text-feedback-conflict/4';
+            router.setUrl(testUrl);
+
+            fixture.detectChanges();
+
+            expect(courseManagementStub).to.have.been.calledWith(1);
```

Comment:
It might be better to change the course id and exercise id between tests to see if they are working with other values.

Reply:
Changing values isn't necessary imo (and wouldn't allow me to reuse as much). Either it can parse numbers or it can't.